### PR TITLE
Added qualified exception to ipam

### PR DESCRIPTION
**Pull request by quadewarren**
JIRA:NCP-1912

### CODE REVIEW

File: quark/tests/functional/test_ipam.py [hunk 1]
```diff
@@ -182,3 +183,43 @@ def test_subnets_get_marked_as_full_retroactively(self):
...
+            sub_mod = db_api.subnet_create(self.context, **subnet)
+            # NOTE(asadoughi): update after cidr constructor has been invoked
+            db_api.subnet_update(self.context,
+                                 sub_mod,
+                                 next_auto_assign_ip=next_ip)
+        yield net_mod
+        with self.context.session.begin():
+            db_api.subnet_delete(self.context, sub_mod)
+            db_api.network_delete(self.context, net_mod)
+
```
Comment (roaet) [inline quark/tests/functional/test_ipam.py]: I see you are attempting to reallocate a single provided ipv4 and letting the ipv6 case run through (i think?). Can you make very similar tests that iterate through the other provided possibilities (just ipv6, both ipv6 and ipv4)?

**Comment by roaet** [comment]
LGTM other than test question

File: quark/tests/functional/test_ipam.py (hunk 1 above)
Comment (quadewarren) [inline quark/tests/functional/test_ipam.py]: Yeah, that makes sense. You are only talking about different test cases and not different IPAM strategies, right?

**Comment by roaet** [comment]
LGTM +1 ⛄

**Comment by xroot88** [comment]
+1

**Pull request merged**